src/autoload.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+spl_autoload_register(function ($className) {
+
+    // Abort here if we do not try to load BTCPayServer namespace.
+    if (strpos($className, 'BTCPayServer') === false) {
+        return;
+    }
+
+    $parts = explode('\\', $className);
+    if (count($parts) === 3) {
```

Comment:
This is a bad idea too. If we have a class with 4 parts it breaks?
Why not keep it simple? Like this?
```
$filePath =  __DIR__ . str_replace('\\', PATH_SEPARATOR, $className).'.php';
```

Reply:
I kinda wanted to limit it to 3 Levels so it does not catch up if we have a client implementation e.g. WooCommerce Plugin using BTCPayServer\WooCommerce\Gateway\PaymentGateway class but yes, maybe it is too error prone if we break that structure in the plugin. So your suggestion is less maintenance so we should do it.

Your suggestion would include BTCPayServer in the path which is not present in the filesystem and I think you meant `DIRECTORY_SEPARATOR`, I will figure something out.

Reply:
Yeah `DIRECTORY_SEPARATOR` is what I meant.
If you fear conflicting namespaces, we cal always add `Greenfield` or `Api` as the 2nd part?

Reply:
No fear, just wanted to trim it down to only check our library files. Yes, adding a namespace just for the API might be a bulletproof solution to this. Not sure if it is worth it to have another parent `Greenfield` or `api` directory for all our subdirs?

Reply:
Nah I guess this is fine.
I had my doubts about the line `if (file_exists($filePath)) {` as it could strain disk IO, but we already have a check for classes starting with `BTCPayServer` and it would only add load if the eCommerce integration has a lot of files.
Guess we are nit-picking at this point. Let's move forward.